velox/docs/index.rst

```diff
@@ -14,3 +14,4 @@ Velox Documentation
     bindings/python/README_generated_pyvelox
     develop
     programming-guide
+    Build metrics <https://facebookincubator.github.io/velox/bm-report/>
```

Comment:
@aditi-pandit These build metrics are similar to the CI conbench results. They don't usually belong to the top-level documentation.
We should probably add a `Building` page under `develop` and add this link there along with other build details/instructions.

Reply:
@majetideepak : I created the PR because the build metrics link isn't navigable from Velox site/docs. So it would be easy to forget unless starred. Maybe a good place for the link is the Velox site as well. I'm fine with whatever you feel is more appropriate.

Reply:
Sounds good @kgpai